test/machi_ap_repair_eqc.erl

```diff
@@ -0,0 +1,568 @@
+%% 4. After commands are executed, remove patition and wait for the chain
+%%    without down nodes nor repairing nodes.
+%% 5. Asserting written results so that each record be read from the
+%%    chain and data be the same with written one.
+
+%% Improvements to-do's
+%% - Use higher concurrency, e.g. 10+
+%% - Random length for binary to write
+%% - Operations other than append, write, trim
+%% - Use checksum instead of binary to save memory
+%% - More variety for partitioning pattern: non-constant failure
+
+-module(machi_ap_repair_eqc).
+
+-ifdef(TEST).
+-ifdef(EQC).
+-compile(export_all).
+-include("machi.hrl").
+-include("machi_projection.hrl").
+-include("machi_verbose.hrl").
+-include_lib("eqc/include/eqc.hrl").
+-include_lib("eqc/include/eqc_statem.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+%% ETS table names
+-define(WRITTEN_TAB,  written).  % Successfully written data
+-define(ACCPT_TAB,    accpt).    % Errors with no harm, e.g. timeout
+-define(FAILED_TAB,   failed).   % Uncategorized errors, when happenes
+                                 % it should be re-categorized to accept or critical
+-define(CRITICAL_TAB, critical). % Critical errors, e.g. double write to the same key
+
+-define(QC_OUT(P),
+        eqc:on_output(fun(Str, Args) -> io:format(user, Str, Args) end, P)).
+
+%% EUNIT TEST DEFINITION
+prop_repair_test_() ->
+    {PropTO, EUnitTO} = eqc_timeout(60),
+    Verbose = eqc_verbose(),
+    {spawn,
+     [{timeout, EUnitTO,
+       ?_assertEqual(
+          true,
+          eqc:quickcheck(eqc:testing_time(
+                           PropTO, ?QC_OUT(noshrink(prop_repair(Verbose))))))}]}.
+
+prop_repair_par_test_() ->
+    {PropTO, EUnitTO} = eqc_timeout(60),
+    Verbose = eqc_verbose(),
+    {spawn,
+     [{timeout, EUnitTO,
+       ?_assertEqual(
+          true,
+          eqc:quickcheck(eqc:testing_time(
+                           PropTO, ?QC_OUT(noshrink(prop_repair_par(Verbose))))))}]}.
+
+%% SHELL HELPERS
+test() -> test(100).
+test(N) -> test(N, true).
+test(N, Verbose) -> quickcheck(numtests(N, noshrink(prop_repair_par(Verbose)))).
+check() -> check(prop_repair_par(true), current_counterexample()).
+
+-record(state, {num,             % Number of FLU servers
+                seed,            % Seed for partition simulator
+                verbose=false,   % Verbose output for debugging
+                flu_names,       % List of FLU names
+                mgr_names,       % List of chain manager names
+                fc_list,         % List of FLU1 proxy clients
+                cr_list}).       % List of CR clients
+
+weight(_S, change_partition) ->  20;
+weight(_S, _)                -> 100.
+
+%% append
+
+append_args(#state{cr_list=CRList}=S) ->
+    [elements(CRList), chunk(), S].
+
+append({_SimSelfName, C}, Bin, #state{verbose=V}=S) ->
+    Prefix = <<"pre">>,
+    Len = byte_size(Bin),
+    Res = (catch machi_cr_client:append_chunk(C, Prefix, Bin, {sec(1), sec(1)})),
```

Comment:
Note: with a modification to change `{sec(1), sec(1)}` to `sec(1)`, tests looks passing. While this kind of test fix is not right, but could be a hint for finding root cause.